lib/ecto/migrator.ex

```diff
@@ -561,7 +561,7 @@ defmodule Ecto.Migrator do
     |> Enum.take_while(&(within_target_version?.(&1, target, direction)))
   end
 
-  defp pending_step(versions, migration_source, direction, count) do
+  defp pending_step(versions, migration_source, direction, count) when is_integer(count) do
```

Comment:
in this func, `Enum.take` already prevents the unexpected behavior, but I think it makes more intuitive to understand what type of value should be used.